### ISSUE TITLE
Please add a few more examples using real dataset (other than sklearn.datasets)

### ISSUE THREAD

**Issue by rezacsedu** [albertbup/deep-belief-network#10]
Hi there,

Thanks for the fantastic work. This is very useful to me. Could you please add few more examples using real dataset other than sklearn.datasets? For example, Gene Expression dataset etc. 

Thanks.

**Comment by albertbup** [comment]
Hey, thanks!

What's the particularity in that dataset? I mean, everything you can store in numpy arrays will work. Notice that sklearn datasets are nothing else than objects containing data as numpy arrays. Thus, you just need to parse your dataset  in whatever format was given and store them into numpy arrays.

Best.

**Comment by rezacsedu** [comment]
Hi,
Thanks for reply me. When I load the MNIST dataset and print the contents I got below output: 

[[  0.   0.   5. ...,   0.   0.   0.]
 [  0.   0.   0. ...,  10.   0.   0.]
 [  0.   0.   0. ...,  16.   9.   0.]
 ..., 
 [  0.   0.   1. ...,   6.   0.   0.]
 [  0.   0.   2. ...,  12.   0.   0.]
 [  0.   0.  10. ...,  12.   1.   0.]]

Do you think that it is vectorized? If so, particularly I need to know how to vectorized a sample of the TCGA dataset. Below is a sample of the TCGA dataset: 

Sample  Chromosome  Start  End  Num_Probes   Segment_Mean
BEDEW_p_TCGA_FFPE_7_13_N_GenomeWideSNP_6_A05_1347396  1  61735  836713  68    -0.0961
BEDEW_p_TCGA_FFPE_7_13_N_GenomeWideSNP_6_A05_1347396  1  836746  3208375  663    0.4065
BEDEW_p_TCGA_FFPE_7_13_N_GenomeWideSNP_6_A05_1347396  1  3212946  16007300   7302    0.0227
BEDEW_p_TCGA_FFPE_7_13_N_GenomeWideSNP_6_A05_1347396  1  16007754  16088317  33    0.4554

The last column is the response column for a classification task. Now could you give me some hints how to parse this type of dataset to make numpy array? Do you think that if I just parse and create a large numpy array then the SupervisedDBNClassification will be able to feed that data? 


Thanks.

**Comment by albertbup** [comment]
Ok, so the input of the network (and for any classifier in sklearn) is a numpy 2d array which dimensions are (nb_samples, nb_features) and another array (nb_samples, nb_responses) for the classification/regression targets. Check sklearn documentation for more information.

However, I see your responses are a continuous value so wouldn't be better if you use SupervisedDBNRegression instead?

Best.

**Comment by naagana** [comment]
blockquote, div.yahoo_quoted { margin-left: 0 !important; border-left:1px #715FFA solid !important; padding-left:1ex !important; background-color:white !important; } Hi Albertbup, If somebody uses your code how can it be cited?
Thank you .


Sent from Yahoo Mail for iPhone


On Friday, March 24, 2017, 18:54, albertbup <notifications@github.com> wrote:


Ok, so the input of the network (and for any classifier in sklearn) is a numpy 2d array which dimensions are (nb_samples, nb_features) and another array (nb_samples, nb_responses) for the classification/regression targets. Check sklearn documentation for more information.

However, I see your responses are a continuous value so wouldn't be better if you use SupervisedDBNRegression instead?

Best.

**Comment by naagana** [comment]
Hi Albert, have you published any work using this code. Second, how does one cite your work if this code is used?
